tests/unit/type/list-common.tcl

```diff
@@ -1,5 +1,5 @@
 # We need a value larger than list-max-ziplist-value to make sure
 # the list has the right encoding when it is swapped in again.
 array set largevalue {}
-set largevalue(ziplist) "hello"
-set largevalue(linkedlist) [string repeat "hello" 4]
+set largevalue(listpack) "hello"
+set largevalue(quicklist) [string repeat "x" 8192]
```

Comment:
that `largevalue` doesn't seem like it's still suitable.
it was born at a time where lists were either a ziplist or a linked-list, so large items were not packed at all.
but now we always use listpack (or plain nodes but that's another story). 

i suppose we're counting on the fact that even if we pass the size of `list-max-listpack-size`, we'll still create listpack based quicklists (with one entry per listpack.
if we keep this `largevalue` approach, let's add a comment that explains why it still does what we want.

Reply:
Maybe we can modify these tests by using the feature of forcing quicklist encoding when `list_max_listpack_size` is 0.

Reply:
you mean instead of trying to control the encoding by using different item sizes, we'll control the encoding by changing the config.
i think that's cleaner approach (although i'd rather set it to `1`, not `0`).

i can also live with the current approach (of creating items big enough so that all listpacks end up with only one item and then as soon as there are two items in the list, it must be quicklist encoded, but if we keep that approach, i think a big comment is needed to explain that hack.

maybe we can leave this change for last, and first finish with all other comments, then you can try this, and if it ends up too messy, we can easily revert it (and be sure we didn't revert any other cleanup or fix).

Reply:
I don't think it's appropriate to use `1`, because in this case the quicklist will become a list with only one entry per quicklist node, which will result in a lot of boundary conditions we won't be able to cover.